### PR TITLE
HTCONDOR-1101 stdout/err merged xfer bug

### PR DESCRIPTION
**Pull request by JaimeFrey**
https://opensciencegrid.atlassian.net/browse/HTCONDOR-1101

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed
- [x] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

**Comment by Todd-L-Miller** [comment]
Good fix to a stupid bug.

Jaime and I talked about testing and it seems like it would probably be both quick and worthwhile to write a test here: the same test can check both that merging the standard streams works correctly and that when we do, we also transfer an output file.

**Comment by JaimeFrey** [comment]
I've added a test.

File: src/condor_tests/job_filexfer_merged-stdout_van.run [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+	if(!-f "$outh"){ print "$outh failed to arrive\n"; exit(1)};
+	if(!-f "$outi"){ print "$outi failed to arrive\n"; exit(1)};
+	if(!-f "$outj"){ print "$outj failed to arrive\n"; exit(1)};
+
+	# If the job's stdout/err are merged properly (with interleaving),
+	# we should see these as the first three lines:
+	# (from stdout):Called with:...
+	# (from stderr):Hello
+	# (from stdout):Working Directory is ...
+	if(!open(FH, '<', "$stdout")){ print "Faild to open job stdout\n"; exit(1)};
```
Comment (Todd-L-Miller) [inline src/condor_tests/job_filexfer_merged-stdout_van.run:73]:
```suggestion
	if(!open(FH, '<', "$stdout")){ print "Failed to open job stdout\n"; exit(1)};
```

**Review by Todd-L-Miller — APPROVED** [review]
Looks good to me.  Thanks.  Feel free to accept the typo suggestion before merging.

**Pull request merged**